src/NServiceBus.Core/Saga/UniqueAttribute_obsolete.cs

```diff
@@ -17,19 +18,10 @@ public sealed class UniqueAttribute : Attribute
         /// </summary>
         /// <param name="type">The <see cref="Type"/> to evaluate.</param>
         /// <returns>A <see cref="PropertyInfo"/> of the property marked with a <see cref="UniqueAttribute"/> or null if not used.</returns>
+        [ObsoleteEx(RemoveInVersion = "7.0", TreatAsErrorFromVersion = "6.0", Message = "Use the new SagaMetadata")]
```

Comment:
should be enough to obsolete the class. unless the methods/props have more explicit alternatives and u want to show a different compile error